### PR TITLE
modified as per request asked for changes in PR

### PR DESCRIPTION
**Pull request by Roshan-Kumar-Sharma**
Done some additional modification in users.events.js, configs.js

### CODE REVIEW

File: api/v1/users/users.controller.js [hunk 1]
```diff
@@ -89,57 +95,99 @@ module.exports.login = async (req, res, next) => {
...
+      tasks: ['email'],
+    },
+  }
+
+  const part1 = req.payload.email.slice(0, 3)
+  const part2 = req.payload.email.split('.')[1]
+  const resEmail = `${part1}*****.${part2}`
+
+  eventEmitter.emit('sendEmailEvent', data, 'verificationEmail')
   res
```
Comment (shubdev07) [inline api/v1/users/users.controller.js:113]: Use responseGenerator

File: api/v1/users/users.controller.js [hunk 2]
```diff
@@ -89,57 +95,99 @@ module.exports.login = async (req, res, next) => {
...
-      if (task === 'email') isValidToChangeEmail = true
-    })
-
-    if (isValidToChangeEmail) {
-      // eslint-disable-next-line no-unused-vars
-      const user = await User.findOneAndUpdate(
-        { email: req.payload.email },
-        { isVerified: true },
-        { new: true }
+    if (!req.payload.metadata.tasks) {
```
Comment (shubdev07) [inline api/v1/users/users.controller.js:144]: what is this doing?

Comment (shubdev07) [inline api/v1/users/users.controller.js:145]: Combine this check with the below check using && operator

File: api/v1/users/users.controller.js [hunk 3]
```diff
@@ -89,57 +95,99 @@ module.exports.login = async (req, res, next) => {
...
+    }
+
+    if (req.payload.metadata.tasks.indexOf('email') === -1) {
       throw createHttpError.Unauthorized(
         'Token is not valid for updating email status'
       )
     }
+
+    // eslint-disable-next-line no-unused-vars
+    const user = await User.findOneAndUpdate(
```
Comment (shubdev07) [inline api/v1/users/users.controller.js:158]: remove eslint disable

**Comment by Roshan-Kumar-Sharma** [comment]
conflicts resolved...

**Pull request merged**